sql/src/test/java/org/apache/druid/sql/calcite/CalciteWindowQueryTest.java

```diff
@@ -91,119 +88,117 @@ public static Object parametersForWindowQueryTest() throws Exception
+    {
+      if (results.exception != null) {
+        throw new RE(results.exception, "Failed to execute because of exception.");
+      }
+      Assert.assertEquals(1, results.recordedQueries.size());
+
+      final WindowOperatorQuery query = getWindowOperatorQuery(results.recordedQueries);
+      for (int i = 0; i < input.expectedOperators.size(); ++i) {
+        final OperatorFactory expectedOperator = input.expectedOperators.get(i);
+        final OperatorFactory actualOperator = query.getOperators().get(i);
+        if (!expectedOperator.validateEquivalent(actualOperator)) {
+          assertEquals("Operator Mismatch, index[" + i + "]",
+              queryJackson.writeValueAsString(expectedOperator),
+              queryJackson.writeValueAsString(actualOperator));
+          fail("validateEquivalent failed; but textual comparision of operators didn't reported the mismatch!");
+        }
+      }
+      final RowSignature outputSignature = query.getRowSignature();
+      ColumnType[] types = new ColumnType[outputSignature.size()];
+      for (int i = 0; i < outputSignature.size(); ++i) {
+        types[i] = outputSignature.getColumnType(i).get();
+        Assert.assertEquals(types[i], results.signature.getColumnType(i).get());
+      }
 
-    Function<Object, String> jacksonToString = value -> {
-      try {
-        return queryJackson.writeValueAsString(value);
+      maybeDumpActualResults(results.results);
+      for (Object[] result : input.expectedResults) {
+        for (int i = 0; i < result.length; i++) {
+          // Jackson deserializes numbers as the minimum size required to
+          // store the value. This means that
+          // Longs can become Integer objects and then they fail equality
+          // checks. We read the expected
+          // results using Jackson, so, we coerce the expected results to the
+          // type expected.
+          if (result[i] != null) {
+            if (result[i] instanceof Number) {
+              switch (types[i].getType()) {
+                case LONG:
+                  result[i] = ((Number) result[i]).longValue();
+                  break;
+                case DOUBLE:
+                  result[i] = ((Number) result[i]).doubleValue();
+                  break;
+                case FLOAT:
+                  result[i] = ((Number) result[i]).floatValue();
+                  break;
+                default:
+                  throw new ISE("result[%s] was type[%s]!?  Expected it to be numerical", i, types[i].getType());
+              }
+            }
+          }
+        }
       }
-      catch (JsonProcessingException e) {
-        throw new RE(e);
+      assertResultsValid(filename, input.expectedResults, results);
+    }
+
+    private void maybeDumpActualResults(List<Object[]> results) throws Exception
+    {
+      for (Object[] row : results) {
+        System.out.println("  - " + queryJackson.writeValueAsString(row));
```

Comment:
since we're rewriting this code - maybe we can write this through a logger

Reply:
I'm afraid that would partially defeat the original purpose of these line ; as in that case it would prefix the results with the timestamp/etc.

so instead
```
  - [1.0,21.0]
  - [2.0,21.0]
  - [3.0,21.0]
  - [4.0,21.0]
  - [5.0,21.0]
  - [6.0,21.0]
```
we would get:
```
2023-10-09T10:57:46,840 INFO [main] org.apache.druid.sql.calcite.BaseCalciteQueryTest -   - [1.0,21.0]
2023-10-09T10:57:46,841 INFO [main] org.apache.druid.sql.calcite.BaseCalciteQueryTest -   - [2.0,21.0]
2023-10-09T10:57:46,841 INFO [main] org.apache.druid.sql.calcite.BaseCalciteQueryTest -   - [3.0,21.0]
2023-10-09T10:57:46,841 INFO [main] org.apache.druid.sql.calcite.BaseCalciteQueryTest -   - [4.0,21.0]
2023-10-09T10:57:46,841 INFO [main] org.apache.druid.sql.calcite.BaseCalciteQueryTest -   - [5.0,21.0]
2023-10-09T10:57:46,841 INFO [main] org.apache.druid.sql.calcite.BaseCalciteQueryTest -   - [6.0,21.0]
```

which would need some preprocessing to get it into the `yaml` file.

I think an alternate way of fixing these kind of issues could be to create a full new testcase file file with the expectations and everything....not sure how easy/hard would that be

Reply:
I've changed this to produce the full results; and log it with a single call :)